views/js/qtiCreator/helper/languages.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015-2022 (original work) Open Assessment Technologies SA ;
+ */
+
+
+/**
+ * Languages utils to handle getting list of available language name, code and direction.
+ * v1 example: {"ar-arb": "Arabic"}
+ * v2(recent) example: {code: "ar-arb", label: "Arabic", orientation: "http://www.tao.lu/Ontologies/TAO.rdf#OrientationRightToLeft", uri: "http://www.tao.lu/Ontologies/TAO.rdf#Langar-arb"}
+ */
+ define([
+    'util/url',
+    'core/dataProvider/request'
+], function (urlUtil, request) {
+    'use strict';
+
+    const languagesUrl = urlUtil.route('index', 'Languages', 'tao');
+    const headers = {'Accept-version': 'v2'};
+
+    let languagesRequest = null;
+    let legacyLanguagesData = null;
+    let CKELanguagesData = null;
+
+    /**
+     * Format languages to v1 format to avoid breaking changes in existing code,
+     * Warning Please avoid usage of this method for new development.
+     * uses legacyLanguagesData as a cache
+     *
+     * @param {*} languages languages v2 format
+     * @returns {Object} languages v1 data format
+     */
+    const useLegacyFormatting = (languages) => {
+        if(!legacyLanguagesData) {
+            legacyLanguagesData = languages.reduce((memo ,lang) => {
+                memo[lang.code] = lang.label;
+
+                return memo;
+            }, {});
+        }
+
+        return legacyLanguagesData;
+    }
```

Comment:
**quibble:** I'm not sure the approach is correct with the caching. A list is given as a parameter, then it is transformed and the result is cached into a private variable. The next time you call the help with another list, it will return the result from the first call...

I'd rather suggest not caching the transformation because performance benefits are limited. Otherwise, the helper would be better off not accepting a parameter, it could act as an alternative to the loader, internally calling the main helper `getList()`.

BTW, the code contains several code-style issues. You'd need to pass it through Prettier.